### ISSUE TITLE
Deno Language Server Error When Importing Local File

### ISSUE THREAD

**Issue by xpyxel** [axetroy/vscode-deno#61]
### Deno Language Server Error When Importing Local File

**Describe the bug**
Attempting to import local files gives an error of ```Could not find module `file` locally.```

**To Reproduce**

1. Install the VSCode extension
2. Create a secondary file named `something.ts`
3. Attempt to import it from your `main.ts` or `index.ts` file

**Expected behavior**

It should not show a error within the editor and acknowledge the file's existence. 

**Screenshots**

![](https://owo.sh/98QLjVZ.png)

**Versions**

vscode: 1.42.1
deno: 0.33.0
extension: 2.0.1

**Comment by axetroy** [comment]
This is a Windows path processing issue.

But unfortunately, I do n’t have a Windows machine nearby to query the problem.

Can you provide the full path to this `logger.ts`?

This should help me troubleshoot

**Comment by xpyxel** [comment]
Full path is `D:\random\logger.ts`, when using the full path it doesn't error
![](https://owo.sh/6MuH5aK.png)

**Comment by axetroy** [comment]
I tried to fix this in v2.0.3

Since there is no Window machine as a test, I am not sure if it has been fixed

If it has been fixed, can you give feedback

**Comment by xpyxel** [comment]
I'll try to see if it's fixed

**Comment by xpyxel** [comment]
Does not seem like it is, from what i can tell.
![](https://owo.sh/9u71GU9.png)

**Comment by axetroy** [comment]
Ham ....

I don't know why it's not fixed

Or wait for a days, I can debug with Window machine

Or can you help me fix it, that's better

The relevant code is here

https://github.com/axetroy/vscode-deno/blob/78e3f59454cdab6fd6b410c7d8f0e40c23263555/server/src/deno.ts#L267-L272

```js
const path = require('path');

const filepath = path.win32.resolve('C:\\random\\', './test.ts'.replace('/', path.win32.sep))

console.log(filepath); // 'C:\\random\\test.ts'
```

**Comment by xpyxel** [comment]
using `console.log` to print `filepath` on line 268 seems it's attempting to resolve it to `d:/d:/` instead of just `d:/`

**Comment by xpyxel** [comment]
adding a replace to `importerFolder` seems to fix it, however i am unsure if it breaks anything on linux / mac.
```ts
        filepath = path.resolve(
          importerFolder.replace("/", ""),
          moduleName.replace(/\//g, path.sep)
        );```

**Comment by InDIOS** [comment]
Hi there, very nice work with this extension, thanks for that. I had fixed this bug with a simple conditional that checks if the `importerFolder` variable has a windows path pattern and i thing that should work in linux as well. is just this:
```ts
// /server/src/deno.ts[268-270]
} else {
+  if (/^\/[a-zA-Z]\:\//.test(importerFolder)) {
+      importerFolder = importerFolder.slice(1);
+  }
   filepath = path.resolve(
     importerFolder,
     moduleName.replace(/\//g, path.sep)
   );
}
```
Good work, I'm waiting for more features soon.
PS: Sorry for my english

**Comment by axetroy** [comment]
@InDIOS Since there is no test machine, I am not sure if this fixes

**Comment by axetroy** [comment]
Hi!

vscode-deno@3.0.0 have been release. can you try it again?

**Comment by xpyxel** [comment]
This issue looks fixed as of version 3.0.0. Closing the issue.
